modulus/launch/utils/checkpoint.py

```diff
@@ -277,8 +282,9 @@ def load_checkpoint(
     scheduler: Union[scheduler, None] = None,
     scaler: Union[scaler, None] = None,
     epoch: Union[int, None] = None,
+    return_metadata: bool = False,
```

Comment:
Instead of returning meta data, can you provide a dictionary object that gets modified? This way we avoid this boolean flag and its more consistent with the other parameters:

meta_data: Union[dict, None] = None,

Then:
```
if meta_data:
  for key, value in checkpoint_dict.get("metadata"):
     meta_data[key] = value
```

Reply:
Return type then doesnt need to change as well

Reply:
+1 to @NickGeneva suggestion. In this case, users won't have to deal with different return types that lead to code like this:
```pyt
r = save_checkpoint(..)
if isinstance(r, tuple):
  # handled tuple
elif ...
```

Reply:
Sounds good. Changed it according to the suggestion